### ISSUE TITLE
States Base color not setting background

### ISSUE THREAD

**Issue by brburnette** [okviz/CardWithStates#21]
The Base color is ONLY applied to the data label even when I have Background selected in the Affect drop down. The Target color does change the background though.

**Comment by tamasys** [comment]
@danieleperilli I can reproduce this issue, the base colour always applies to the text regardless of what behaviour is selected. See attached report file.
[Example of background colour issue.zip](https://github.com/okviz/CardWithStates/files/2368939/Example.of.background.colour.issue.zip)

**Comment by danieleperilli** [comment]
If you are talking about the first two visual in your file: when states are applied to background, the text color is automatically changed according to an acceptable level of contrast. 
If you are talking about the third one: I suggest you to read this article to understand how states work: https://okviz.com/blog/states-in-custom-visuals

**Comment by tamasys** [comment]
The issue is consistency between the colour selectors: In that example file, the three visuals are identical, the only difference is the number displayed on each one. Base is set to red, Target 1 to yellow, and Target 2 to green.

The yellow and green work as expected - they colour the background when the threshold is reached. But the red doesn't behave like that, it colours the text and leaves the background white, even though "Background" is the selected behaviour.
![states](https://user-images.githubusercontent.com/2321397/45346611-f9a43e00-b5ec-11e8-8e55-9b23758c44a1.PNG) ![cards4](https://user-images.githubusercontent.com/2321397/45346786-96ff7200-b5ed-11e8-9108-1f93eb6c46cf.PNG)

**Comment by danieleperilli** [comment]
Yes, now understand the issue. If the base color is applied, the affect option is ignored. It seems a bug. Thanks for explained it.

**Comment by tamasys** [comment]
@raulrdodriguez This looks like an unrelated issue that you might want to raise a separate issue for if you think there's a bug. It sounds like you're trying to have different colours for positive and negative variance. What I'd suggest you try is
1. Create a **Variance %** measure calculating the variance
2. Add the **Variance %** measure to _States Measure_ field
3. Add the state values to the _States_ field as usual
4. Use absolute values in _Value Type_

It should now change colour based on positive and negative variance instead of the absolute variance.

**Comment by raulrdodriguez** [comment]
@tamasys thanks for your feedback.
I don't have a Variance % column in my data, I do have a "Target" and an "Actual" column. Then I only turn ON the "Variance" option and select "Percentage".
Are the states not going to work with the automatic powerbi Variance calculation? Do I need to have that column pre-calculated in order to have the color states I need?

Thanks again!

Raul

**Comment by danieleperilli** [comment]
Raul, the variance is calculated by the visual. Please, provide a repro file by attaching to this post or by submitting to support@okviz.com.

**Comment by raulrdodriguez** [comment]
Report attached. Thanks Daniele
[PowerBI Production HRs Test.zip](https://github.com/okviz/CardWithStates/files/2388600/PowerBI.Production.HRs.Test.zip)
